components/style/gecko_conversions.rs

```diff
@@ -34,3 +34,41 @@ impl From<nsStyleCoord_CalcValue> for CalcLengthOrPercentage {
         }
     }
 }
+
+impl From<LengthOrPercentage> for nsStyleCoord_CalcValue {
+    fn from(other: LengthOrPercentage) -> nsStyleCoord_CalcValue {
+        match other {
+            LengthOrPercentage::Length(au) => {
+                nsStyleCoord_CalcValue {
+                    mLength: au.0,
+                    mPercent: 0.0,
+                    mHasPercent: false,
+                }
+            },
+            LengthOrPercentage::Percentage(pc) => {
+                nsStyleCoord_CalcValue {
+                    mLength: 0,
+                    mPercent: pc,
+                    mHasPercent: true,
+                }
+            },
+            LengthOrPercentage::Calc(calc) => calc.into(),
+        }
+    }
+}
+
+impl From<nsStyleCoord_CalcValue> for LengthOrPercentage {
+    fn from(other: nsStyleCoord_CalcValue) -> LengthOrPercentage {
+        match (other.mHasPercent, other.mLength) {
+            (false, _) => LengthOrPercentage::Length(Au(other.mLength)),
+            (true, 0) => LengthOrPercentage::Percentage(other.mPercent),
+            _ => {
+                let calc = CalcLengthOrPercentage {
+                    length: Some(Au(other.mLength)),
+                    percentage: None,
+                };
+                LengthOrPercentage::Calc(calc)
```

Comment:
My code is also wrong, I see (should be `other.mPercent`, not `None`, for the percentage value).